src/content/docs/extensions/llm.mdx

```diff
@@ -0,0 +1,232 @@
+
+- **prompt**: The input text to embed.
+- **provider**: The embedding provider, e.g., `'open-ai'`, `'ollama'`, `'voyage-ai'`, etc.
+- **model**: The identifier of the embedding model. This may change overtime; reference the provider's API reference before use.
+- **dimensions** *(optional)*: The size of the embedding vector to be returned, if supported by the provider. Must be a positive integer.
+- **region** *(optional)*: A region or endpoint hint, if supported by the provider.
+
+### Examples
+
+```sql
+CREATE_EMBEDDING("Hello world", "openai", "text-embedding-3-small")
+```
+
+```sql
+CREATE_EMBEDDING("Hello world", "ollama", "nomic-embed-text")
+```
+
+```sql
+CREATE_EMBEDDING("Hello world", "google-gemini", "gemini-embedding-exp-03-07")
+```
+
+With optional parameters:
+
+```sql
+CREATE_EMBEDDING("Hello world", "voyage-ai", "voyage-3-large", 512)
+```
+
+```sql
+CREATE_EMBEDDING("Hello world", "amazon-bedrock", "amazon.titan-embed-text-v1", "us-east-1");
+```
+
+```sql
+CREATE_EMBEDDING("Hello world", "google-vertex", "gemini-embedding-001", 256, "us-east1");
+```
+
+---
+
+### Dimensions Support
+
+Dimension configuration is supported for the following providers.
+
+* open-ai
+* google-vertex
+* voyage-ai
+
+:::note[Note]
+The list of models that support dimension configuration may change
+overtime. The set of valid dimensions is determined by the provider and may also
+change over time. It is highly recommended to check the provider's API reference
+before use. If an unsupported dimension is specified, the provider's API will return an error.
+:::
+---
+
+## Synergy with Vector Extension
+The following example uses the same dataset as the example in the Vector
+documentation. Here we use the embedding function to fetch embeddings.
+
+```python
+# create_embeddings.py
+import pyarrow
+import polars
+import kuzu
+
+DB_NAME = ""
+
+# Initialize the database
+db = kuzu.Database(DB_NAME)
+conn = kuzu.Connection(db)
+
+# Install and load vector extension
+conn.execute("INSTALL vector; LOAD vector;")
+conn.execute("INSTALL llm; LOAD llm;")
+
+
+# Create tables
+conn.execute("CREATE NODE TABLE Book(id SERIAL PRIMARY KEY, title STRING, title_embedding FLOAT[1536], published_year INT64);")
+conn.execute("CREATE NODE TABLE Publisher(name STRING PRIMARY KEY);")
+conn.execute("CREATE REL TABLE PublishedBy(FROM Book TO Publisher);")
+
+# Sample data
+titles = [
+    "The Quantum World",
+    "Chronicles of the Universe",
+    "Learning Machines",
+    "Echoes of the Past",
+    "The Dragon's Call"
+]
+publishers = ["Harvard University Press", "Independent Publisher", "Pearson", "McGraw-Hill Ryerson", "O'Reilly"]
+published_years = [2004, 2022, 2019, 2010, 2015]
+
+# Insert sample data - Books with embeddings
+for title, published_year in zip(titles, published_years):
+    conn.execute(
+        """CREATE (b:Book {title: $title, title_embedding: create_embedding($title, 'open-ai', 'text-embedding-3-small'), published_year: $year});""",
```

Comment:
Point out in comments that the embeddings are being created in cypher instead of in Python.